plugin/octant.d.ts

```diff
@@ -113,31 +113,35 @@ export interface DashboardClient {
   /**
    * Get attempts to fetch a resource using the key provided
    * @param key - the key of the object to be fetched
+   * @param metadata - Optional arbitrary contexual data sent to octant
    * @throws will throw an exception if there is an error with the request
    */
-  Get(key: Key): any;
+  Get(key: Key, metadata?: Record<string, any>): any;
```

Comment:
Octant will serialize these records when it invokes the plugin's `Get` implementation, either via the gRPC interface or via the JS execution environment, correct?

I wonder if we should use a more constrained type for the values than `any`. I worry about encountering edge cases when serializing complicated data types over both of these interfaces (e.g. differences in numeric representations between gRPC and JS, or cycle detection in serialized object graphs, etc).

If we started off by using just a `Record<string, string>`, we could always expand the type in the future. Clawing it back if we encounter edge cases with `any` will be backwards-incompatible.